src/commands/register.ts

```diff
@@ -119,10 +121,36 @@ export default class ComponentRegister extends Command {
       config: raw_config,
     };
 
+    const component_name = ComponentSlugUtils.parse(raw_config.name).component_name;
+    let previous_config_data;
+    try {
+      previous_config_data = await this.getComponentConfig(selected_account.name, component_name, tag);
+    /* eslint-disable-next-line no-empty */
+    } catch {}
+
     cli.action.start(chalk.blue(`Registering component ${raw_config.name}:${tag} with Architect Cloud...`));
     await this.app.api.post(`/accounts/${selected_account.id}/components`, component_dto);
     cli.action.stop();
     this.log(chalk.green(`Successfully registered component`));
+
+    this.log(chalk.blue(`Begin component config diff`));
```

Comment:
Can you put the diff before:
```ts
cli.action.start(chalk.blue(`Registering component ${raw_config.name}:${tag} with Architect Cloud...`));
```

Reply:
Didn't you want the config to compare to be pulled from the API? If we're going to pull the new config to use after it's pushed, we have to register first

Reply:
Why do we need to pull the new config? We post the new config to register the component so we should already have it?

Reply:
Never mind, I see what you mean